modules/miner/miner.go

```diff
@@ -194,18 +210,21 @@ func New(cs modules.ConsensusSet, tpool modules.TransactionPool, w modules.Walle
 // Close terminates all ongoing processes involving the miner, enabling garbage
 // collection.
 func (m *Miner) Close() error {
+	if err := m.tg.Stop(); err != nil {
+		return err
+	}
+
 	m.mu.Lock()
 	defer m.mu.Unlock()
```

Comment:
oh, also I think the miner lock is just a normal lock. No readlock because the miner uses the lock a lot and I wanted it to be fast.